chaoscenter/authentication/api/handlers/rest/project_handler.go

```diff
@@ -271,6 +271,15 @@ func CreateProject(service services.ApplicationService) gin.HandlerFunc {
 			c.JSON(utils.ErrorStatusCodes[utils.ErrInvalidRequest], presenter.CreateErrorResponse(utils.ErrInvalidRequest))
 			return
 		}
+
+		// admin/user shouldn't be able to perform any task if it's default pwd is not changes(initial login is true)
+		initialLogin, err := CheckInitialLogin(service, userRequest.UserID)
```

Comment:
How about adding these checks in the middleware. We anyways need to restrict all the operations till the password is updated. We also have to restrict access to GQL APIs

Reply:
Update: No need to add the check in the middleware, instead checks can be added in operations like create project, create user etc which will also block other subsequent operations. 
Adding checks in these functions will also not require the checks to be added in gql APIs since gql APIs need a project and create project requires password updation.